src/Symfony/Component/Yaml/Parser.php

```diff
@@ -133,6 +133,10 @@ public function parse($value, $exceptionOnInvalidType = false, $objectSupport =
                     throw $e;
                 }
 
+                if (is_float($key)) {
```

Comment:
don't know if it's really necessary, but what about adding a small comment:

```php
// convert float keys to strings, to avoid being converted to integers by PHP
if (is_float($key)) {
    $key = (string) $key;
}
```